index.js

```diff
@@ -4,14 +4,29 @@ var util = require('util'),
     config = require('config');
 
 
-config.util.substituteEnv = function (config) {
-    var decoratedEnvironment =
-        _.object(_(process.env).map(function (v, k) {
-            return [util.format("${%s}", k), v];
-        }));
-    config.util.substituteDeep(decoratedEnvironment, config);
+var substituteEnv = function (config) {
+
+    function _recur(obj) {
+        _(obj).forEach(function (v, k) {
+            if (_.isString(v)) {
+                var match = v.match('^[$]{(.*)}$');
+
+                if (match) {
+                    if (! process.env[match[1]]) {
+                        throw new Error(util.format('misty-config: %s not set in the environment', v));
+                    } else {
+                        obj[k] = process.env[match[1]];
+                    }
+                }
+            } else if (_.isObject(v) || _.isArray(v)) {
+                _recur(v);
+            }
+        });
+    }
+    _recur(config);
 };
-config.util.substituteEnv(config);
+
+substituteEnv(config);
 config.util.makeImmutable(config);
```

Comment:
Coolies, I just remembered a few days back when we were calling that so just wanted to double-check. :)